wp-admin/js/customizer.js

```diff
@@ -0,0 +1,70 @@
+(function () {
+	wp.customize.bind('ready', function () {
+		let siteIcon = wp.customize('site_icon').get();
+
+		/**
+		 * Listens to icon update. This includes following scenarios.
+		 *
+		 * 1. Icon is removed.
+		 * 2. Icon is updated.
+		 * 3. Icon set as maskable.
+		 * 4. Icon set as un-maskable.
+		 */
+		const iconUpdateListener = function () {
+			siteIcon = parseInt(this.iconId, 10);
+
+			// Check/uncheck maskable checkbox.
+			wp.customize('pwa_maskable_icon').set(this.checked);
+
+			if (!siteIcon) {
+				//alert( wp.customize.control( 'pwa_maskable_icon' ) );
+				wp.customize.control('pwa_maskable_icon').deactivate();
+				return;
+			}
+
+			// At this point we are sure that icon is set, thus activate control.
+			wp.customize.control('pwa_maskable_icon').activate();
+
+			const iconPreview = document.querySelector('img.app-icon-preview');
+
+			if (iconPreview) {
+				document.querySelector('img.app-icon-preview').style.clipPath =
+					this.checked && siteIcon ? 'inset(10% round 50%)' : '';
```

Comment:
The `siteIcon` check isn't needed because if it is not set then the `img` won't be in the document in the first place. Also, instead of `this.checked` you can just use `wp.customize('pwa_maskable_icon').get()`:

```suggestion
					wp.customize('pwa_maskable_icon').get() ? 'inset(10% round 50%)' : '';
```